File: Hackerrank/30 Days of Code/Day14_Scope.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

class Difference {
    private:
    vector<int> elements;
  
  	public:
  	int maximumDifference;
	
	// Add your code here
    Difference(vector<int> elms){
        maximumDifference = 0;
        elements = elms;
    }
    int computeDifference(){
        int diff = 0;
        int n = elements.size();
        for(int i=0; i<n; i++){
            for(int j=i+1; j<n; j++){
                diff = abs(elements[i] - elements[j]);
                if(maximumDifference < diff)
                    maximumDifference = diff; 
            }
        }
        return maximumDifference;
    }
	
}; // End of Difference class

int main() {
    int N;
    cin >> N;
    
    vector<int> a;
    
    for (int i = 0; i < N; i++) {
        int e;
        cin >> e;
        
        a.push_back(e);
    }
    
    Difference d(a);
    
    d.computeDifference();
    
    cout << d.maximumDifference;
    
    return 0;
}

File: Hackerrank/30 Days of Code/Day14_Scope.py
class Difference:
    def __init__(self, a):
        self.__elements = a
		
	# Add your code here
    def computeDifference(self):
        self.maximumDifference = 0
        n = len(self.__elements)
        for i in range(n):
            for j in range(1+i, n):
                diff =  abs(self.__elements[i] - self.__elements[j])
                if(self.maximumDifference < diff):
                    self.maximumDifference = diff
        
		return self.maximumDifference


# End of Difference class

_ = input()
a = [int(e) for e in input().split(' ')]

d = Difference(a)
d.computeDifference()

print(d.maximumDifference)